.github/workflows/nightly-browser-view.yml

```
@@ -0,0 +1,229 @@
+      - name: nightly/upload-build
+        uses: actions/upload-artifact@0b7f8abb1508181956e8e162db84b466c27e18ce # v3.1.2
+        with:
+          name: build
+          path: ./build
+          retention-days: 5 ## No need to keep them since they are uploaded on S3
+
+  upload-to-s3:
+    runs-on: ubuntu-22.04
+    outputs:
+      links: ${{ steps.generate-linklist.outputs.linklist }}
+    needs:
+      - build-mac-installer
+      - build-msi-installer
+      - build-linux
+    steps:
+      - name: nightly/setup-aws-credentials
+        uses: aws-actions/configure-aws-credentials@67fbcbb121271f7775d2e7715933280b06314838 # v1.7.0
+        with:
+          aws-region: us-east-1
+          aws-access-key-id: ${{ secrets.MM_DESKTOP_RELEASE_AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.MM_DESKTOP_RELEASE_AWS_SECRET_ACCESS_KEY }}
+      - name: nightly/download-builds
+        uses: actions/download-artifact@9bc31d5ccc31df68ecc42ccf4149144866c47d8a # v3.0.2
+      - name: nightly/setup-files-for-aws
+        run: |
+          mkdir -p ./aws-s3-dist
+          cp -r --backup=numbered ./build/{macos-release,win-release,linux}/* ./aws-s3-dist
+      - name: nightly/upload-to-s3
+        run: aws s3 cp ./aws-s3-dist/ s3://releases.mattermost.com/desktop/ --acl public-read --cache-control "no-cache" --recursive
+      - name: nightly/generate-linklist
+        id: generate-linklist
+        run: |
+          mkdir -p ./links
+          echo "### Nightly builds:" > ./links/linklist.txt
+          echo "Links for $(date +"%b-%d-%Y")" >> ./links/linklist.txt
+          echo "##### :tux: Linux" > ./links/linklist.txt
+          for i in `ls ./build/linux/$(jq -r .version package.json)/` ; do echo "- [$i](https://s3.amazonaws.com/releases.mattermost.com/desktop/$(jq -r .version package.json)/$i)" ; done >> ./links/linklist.txt
+          echo "##### :apple_logo: macOS" >> ./links/linklist.txt
+          for i in `ls ./build/macos-release/$(jq -r .version package.json)/` ; do echo "- [$i](https://s3.amazonaws.com/releases.mattermost.com/desktop/$(jq -r .version package.json)/$i)" ; done >> ./links/linklist.txt
+          echo "##### :windows: Windows" >> ./links/linklist.txt
+          for i in `ls ./build/win-release/$(jq -r .version package.json)/` ; do echo "- [$i](https://s3.amazonaws.com/releases.mattermost.com/desktop/$(jq -r .version package.json)/$i)" ; done >> ./links/linklist.txt
+          cat ./links/linklist.txt
+          echo "linklist=$(<./links/linklist.txt)" >> "$GITHUB_OUTPUT"
+
+  share-links-to-channel:
+    runs-on: ubuntu-22.04
+    needs:
+      - upload-to-s3
+    steps:
+      - name: nightly/share-links-to-channel
+        run: |
+          echo '{}' | jq "{
+            \"username\": \"NightBuilder\",
+            \"icon_url\": \"https://upload.wikimedia.org/wikipedia/commons/1/17/Luna_symbol.png\",
+            \"text\": \"${{ needs.upload-to-s3.outputs.links }}\"
+          }" >> /tmp/webhook-data.json
```

Building Json under Bash can be error prone. 

Maybe consider replacing this with: 
```suggestion
      jq --null-input \
              --arg icon_url "https://upload.wikimedia.org/wikipedia/commons/1/17/Luna_symbol.png" \
              --arg username "NightBuilder" \
              --arg text "${{ needs.upload-to-s3.outputs.links }}" \
              '{"username":$username,"icon_url": $icon_url, "text": $text }' > /tmp/webhook-data.json
```